### PR TITLE
minor optimization

### PR DESCRIPTION
**Pull request by nestornillo**
one ld instruction less
fixed comment

### CODE REVIEW

**Comment by lronaldo** [comment]
Seems quite correct. Nice one, @nestornillo.

Could you please test this change recompiling examples with this new version and testing all ones that do use cpct_drawSolidBox to ensure it works as expected?

If all works well, I will just merge your contribution straight away.

Thank you very much, man :)

**Comment by nestornillo** [comment]
Hola, que pruebas quieres que haga?

**Comment by nestornillo** [comment]
En el juego que estoy preparando (en el que tengo cpct_drawSolidBox en un fichero .s fuera de cpctelera) no hay diferencia de resultado entre las dos versiones, sólo que mi propuesta ocupa un byte menos. Estoy recompilando cpctelera con esto a ver que tal.
De todas formas lo que también faltaría cambiar son los tiempos que tarda la rutina, eso no sé calcularlo

**Comment by lronaldo** [comment]
The calculations are easy. It takes 1 microsecond less  and occupies 1 byte less. The only question is doing a proper testing to ensure that no bugs are inadvertedly introduced with the change. It seems to me that the change is okay, but I prefer to test to ensure there is no problem that escapes to our intuition :)
------------------------------------------
Los cálculos son fáciles. Se tarda 1 microsegundo menos y ocupa 1 byte menos. La única cuestión es hacer una prueba adecuada para asegurarse de que no se introducen bugs sin querer con el cambio. Me parece que el cambio está bien, pero prefiero hacer pruebas para asegurarnos de que no hay ningún problema que se escape a nuestra intuición :)

**Comment by nestornillo** [comment]
He recompilado cpctelera y funciona correctamente, tanto llamando la funcion desde c como ensamblador, pero me he dado cuenta que la última linea del cbinding es "ld a,l" y la primera linea de mi propuesta es "ld l,a", con lo que todavía se puede optimizar más la llamada desde c si se pasa el "ld l,a" de la rutina al asmbinding, y se quita el "ld a,l" del cbindings.  Me miraré lo de los tiempos, haré más pruebas y haré un pull request de los 3 ficheros.

**Comment by lronaldo** [comment]
Yes, you are completely right. `ld a, l` from cbindings can be removed, and `ld l, a` added to asmbindings. With that change, the main asm file can be changed as follows: 
```
; DrawSolidBox code starts here (immediately after getting parameters from stack)
;   ld     h, d             ;; [1] / HL = DE  
;   ld     l, e             ;; [1] \ HL Points to Video Memory, and DE holds a copy which won't be modified
;   ld     e, a             ;; [1] E=A (Colour pattern)
   
   ;; Modify code using width to jump to ensure
   ;; that only width bytes are copied each line 
   ld     a, #128          ;; [2] We need to jump 128 bytes (64 COPY2HL_n_INC*2 bytes) minus the width of the sprite * 2 (2B)
   sub    c                ;; [1]  to do as much COPY2HL_n_INC as bytes the Sprite is wide
   sub    c                ;; [1]
   ld (_jr_offset), a      ;; [4] Modify JR data to create the jump we need
   
   ld     c, l             ;; [1] C=L (Colour pattern)
   ld     h, d             ;; [1] / HL = DE  
   ld     l, e             ;; [1] \ HL Points to Video Memory, and DE holds a copy which won't be modified
;   ld     e, l             ;; [1] Restore DE = HL
```
That represents a huge saving. 3 bytes and 3 microseconds for C bindings, and 1 byte + 1 microsecond for asm bindings. 

Really nice review. If tested it with all CPCtelera examples with no problem. So, please, do this final review and add it to the pull request, then I will accept it right away.

Thank you very much for your contribution :)

File: cpctelera/src/sprites/cpct_drawSolidBox.asm [hunk 1]
```diff
@@ -123,9 +123,7 @@
 .endm
 
 ;; DrawSolidBox code starts here (immediately after getting parameters from stack)
-   ld     h, d             ;; [1] / HL = DE  
-   ld     l, e             ;; [1] \ HL Points to Video Memory, and DE holds a copy which won't be modified
-   ld     e, a             ;; [1] E=A (Colour pattern)
+   ld     l, a             ;; [1] L=A (Colour pattern)
```
Comment (lronaldo) [inline cpctelera/src/sprites/cpct_drawSolidBox.asm:126]: This line should now be removed, as cbindings already have colour pattern in L, and asmbindings copy it from A.

**Comment by nestornillo** [comment]
Is the new time measurement "16+11+ (17+4W)H+11HH" ?  I don't understand where "4 + (17 + 4W)H + 11HH" comes from

**Comment by nestornillo** [comment]
16=c bindings,11=set jump. What was 4?

**Comment by nestornillo** [comment]
Ok, my brain cells don't work well, now I get "18+(17+4W)H+11HH"
18+(17+4W)H+11HH
16+11+(3+4W+3)H+11(H-1)+11HH+2
16[cbindigs] + 11[set jp]+ H * ( 3[jr_0]+4W[copymacro]+3[check last line] ) + (H-1) * 11 [set next line] + HH * 11 [correct next line] + 2 [ret z last line]
I still don't get what was the 4

**Comment by nestornillo** [comment]
My tests match exactly with the "18+(17+4W)H+11HH" formula, I'll make the pull request then

**Comment by nestornillo** [comment]
I don't have much experience with Github. All 3 files are ok now, Should I do something now?

**Comment by nestornillo** [comment]
Umm, now a 1 byte possible optimization would be to remove the "ld l,a" from asmbindings, and change the specifications of cpct_drawSolidBox_asm to say that it recieves the colour pattern in L instead of A. But it would break compatibility with existing code. What do you think?

**Comment by lronaldo** [comment]
> My tests match exactly with the "18+(17+4W)H+11HH" formula, I'll make the pull request then

This expression is correct now for C-bindings. The previous 4 was the counting without bindings, which was performed before CPCtelera standarized C-bindings as the preferred count. On asm now we save only 15 microseconds due to the added and removed `ld l,a` and `ld a, l`. Byte count is also correct (167 / 163).



> Umm, now a 1 byte possible optimization would be to remove the "ld l,a" from asmbindings, and change the specifications of cpct_drawSolidBox_asm to say that it recieves the colour pattern in L instead of A. But it would break compatibility with existing code. What do you think?

Breaking API compatibility is not an option for CPCtelera 1.5. Other functions could also benefit from breaking API, but that would drive many developers into troubles with their projects and compatibility problems on old projects being recompiled. So, this is not considered by now. API compatibility will be considered again for CPCtelera 2.0.

**Pull request merged**